examples/snapshot-metadata-verifier/main.go

```diff
@@ -0,0 +1,177 @@
+	}
+
+	stringFlag(&args.Namespace, "namespace", "n", "", "The Namespace containing the VolumeSnapshot objects.")
+	stringFlag(&args.SnapshotName, "snapshot", "s", "", "The name of the VolumeSnapshot for which metadata is to be displayed.")
+	stringFlag(&args.PrevSnapshotName, "previous-snapshot", "p", "", "The name of an earlier VolumeSnapshot against which changed block metadata is to be displayed.")
+	stringFlag(&args.SourceDevicePath, "source-device-path", "src", "", "The source device to use for verification.")
+	stringFlag(&args.TargetDevicePath, "target-device-path", "tgt", "", "The target device to use for verification.")
+
+	if home := homedir.HomeDir(); home != "" {
+		flag.StringVar(&kubeConfig, "kubeconfig", filepath.Join(home, ".kube", "config"), "Path to the kubeconfig file.")
+	} else {
+		flag.StringVar(&kubeConfig, "kubeconfig", "", "Path to the kubeconfig file.")
+	}
+
+	flag.StringVar(&args.SAName, "service-account", "", "ServiceAccount used to create a security token. If unspecified the ServiceAccount of the Pod in which the command is invoked will be used.")
+	flag.StringVar(&args.SANamespace, "service-account-namespace", "", "Namespace of the ServiceAccount used to create a security token. If unspecified the Namespace of the Pod in which the command is invoked will be used.")
+
+	flag.Int64Var(&args.TokenExpirySecs, "token-expiry", 600, "Expiry time in seconds for the security token.")
+	flag.Int64Var(&args.StartingOffset, "starting-offset", 0, "The starting byte offset.")
+
+	var maxResults int
+	flag.IntVar(&maxResults, "max-results", 0, "The maximum results per record.")
+
+	var showHelp bool
+	flag.BoolVar(&showHelp, "h", false, "Show the full usage message.")
+
+	progName := filepath.Base(os.Args[0])
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "\n"+shortUsageFmt, progName)
+	}
+
+	if len(os.Args) > 1 {
+		flag.Parse()
+		args.MaxResults = int32(maxResults)
+	} else {
+		fmt.Fprintf(os.Stderr, "Missing required arguments\n")
+		fmt.Fprintf(os.Stderr, shortUsageFmt, progName)
+		os.Exit(1)
+	}
+
+	if showHelp {
+		fmt.Fprintf(os.Stderr, usageFmt, progName)
+		flag.PrintDefaults()
+		os.Exit(0)
+	}
+}
+
+func main() {
+	// parse flags and set the output emitter
+	parseFlags()
+
+	// get the K8s config from either kubeConfig, in-cluster or default
+	config, err := buildConfig(kubeConfig)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error loading kubeconfig %s: %v\n", kubeConfig, err)
+		os.Exit(1)
+	}
+
+	clients, err := iterator.BuildClients(config)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error creating clients: %v\n", err)
+		os.Exit(1)
+	}
+
+	args.Clients = clients
+
+	sourceDevice, err := os.Open(args.SourceDevicePath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to open source device %s: %q", args.SourceDevicePath, err)
+		os.Exit(1)
+	}
+	defer sourceDevice.Close()
+
+	targetDevice, err := os.OpenFile(args.TargetDevicePath, os.O_RDWR, 0644)
+	if err != nil {
```

Comment:
The usage doesn't make it clear whether this target file should exist or not. Is `os.O_CREAT` needed?